### ISSUE TITLE
BAD_LOGIN_DATA - deb package Ubuntu 18.04

### ISSUE THREAD

**Issue by thehijacker** [pcloudcom/console-client#130]
Hello,

I kept getting BAD_LOGIN_DATA error. Password is 100% correct. I used pcloudcc_2.1.0-1_amd64_ubuntu_18.04.deb package on Ubuntu 18.04.

Using command:

pcloudcc -u email@domain.com -p -s

How to debug?

Thank you and best regards,
Andrej

**Comment by thehijacker** [comment]
I think I found answer to my issue in this ticket for Python API implementation:

https://github.com/tomgross/pycloud/issues/29

My files are stored in Europe servers. API URL for them is different. Instead of https://api.pcloud.com it should use https://eapi.pcloud.com.

When I fixed the URL in Python pcloud API it I was able to login from it. I tried changing all URL addresses in this client and recompile but I still get BAD_LOGIN_DATA error. So there must be other issues yet to be fixed.

Can the one maintaining this tree fix this please? I am thinking on purchasing 2TB plan so I can sync all my photos from NAS server but I need a working CLI client.

Thank you.

**Comment by blackjackshellac** [comment]
Do you use 2FA for auth?  That has prevented me from using the client.

They don't do much on this client any more, after three years I've pretty much given up hope that they'll ever continue with this command line client.

**Comment by thehijacker** [comment]
Nope. No 2FA for authorization. Just username and password.

Is there another solution to sync a folder from Linux server to pCloud?

**Comment by thehijacker** [comment]
Gave up on pCloud. Ordered lifetime 1TB on koofr and using rclone to sync my folder directly from ubuntu,

27.744G / 49.761 GBytes, 56%, 6.474 MBytes/s, ETA 58m2s (xfr#8182/18194) 

Great upload speed.

**Comment by blackjackshellac** [comment]
Yes, it's a shame about the lack of a command line client, it's my only complaint and the one that prevents me from recommending it to any of my colleagues.

Thanks for the update on rclone, that looks interesting.  Apparently it supports pcloud.

**Comment by FransTheekrans** [comment]
I'm having the same issue and it is frustrating to see this has been open already for months. @pcloudcom : When will you solve this? I'd like to use this command line client from europe!

**Comment by FransTheekrans** [comment]
Friendly reminder

**Comment by FransTheekrans** [comment]
friendly reminder

**Comment by PGRHRDS** [comment]
I fixed this by editing pCloudCC/lib/pclsync/psettings.h
Change `#define PSYNC_API_HOST     "binapi.pcloud.com"`
to `#define PSYNC_API_HOST     "bineapi.pcloud.com"`
and remake pcloudcc

**Comment by coxackie** [comment]
@PGRHRDS if this is confirmed to work, maybe it would be good to open a PR. If you nudge sufficiently through customer care to get the devs involved, they eventually do something. But, guys, do not expect that they are following the threads and issues here. The company is entirely interested only in the electron client.

**Comment by PGRHRDS** [comment]
@coxackie I opened a support ticket with customer care and I will indeed stress the need for a codechange. Good suggestion! Low hopes though....

**Comment by PGRHRDS** [comment]
Support says that the console client is unsupported hence they can't/won't assist me on this. I was offered a transfer from EU to US servers however because the issue "may be related to having an EU account".

**Comment by PGRHRDS** [comment]
I have kindly asked support if the devs might be interested to look at my small fix and update the code.

**Comment by coxackie** [comment]
@PGRHRDS yes, this is how I deal with it as well. I ask them kindly to let the devs know and take a look at the request. I stress the importance f keeping paying Linux customers with minimal care, as we indeed ask for. The devs are surely competent - it is just that the company has decided to only invest in the electron app development.

**Comment by Ratteler** [comment]
> I fixed this by editing pCloudCC/lib/pclsync/psettings.h
> Change `#define PSYNC_API_HOST "binapi.pcloud.com"`
> to `#define PSYNC_API_HOST "bineapi.pcloud.com"`
> and remake pcloudcc

I deleted everything on my Raspberry Pi4 and made this change, Now I get "init failed" when I run pCloudCC.

**Comment by PGRHRDS** [comment]
@Ratteler: seems unrelated to me. It does look like the issue (with solutions) described here: https://github.com/pcloudcom/console-client/issues/86

**Comment by PGRHRDS** [comment]
And wat do you know, dev is interested in my solution. I will hand it to them and hopefully thay are able to fully test it and do a PR.

**Comment by coxackie** [comment]
see? they are not half bad. they are actually good, in my experience. they just need to know that we care and exist   ;)

**Comment by Ratteler** [comment]
> @Ratteler: seems unrelated to me. It does look like the issue (with solutions) described here: #86

Thanks for the reply. Sadly still no love.
I suspect it has something to do with libfuze-dev as I didn't delete or reinstall that.

**Comment by alworx** [comment]
Seen this issue too late and did the same thing as @PGRHRDS : https://github.com/alworx/console-client/commit/ad8427118c9ce1f14465700a28c248fc8ebe2442
But i guess hardcoded EU servers will not be merged in a pull request as this just moves the problem to US clients, a cli parameter for the url would be a more flexible approach.

Starting an EU branch would maybe suffice and takes 2 minutes of developers time, maybe this would be a good start @pcloudcom ?

**Comment by joemat** [comment]
I was running into the same isse, thanks to @alworx  and @PGRHRDS for their analysis and initial commits.
Derived on their work I created [another pull-requests](https://github.com/pcloudcom/console-client/pull/133) which doesn't break the client for US clients but uses an environment variable to switch between both regions.

The best solution would of cause having a cli parameter, but to add this I needed more time to get into the code.

**Comment by MomchilKolev** [comment]
> I was running into the same isse, thanks to @alworx and @PGRHRDS for their analysis and initial commits.
> Derived on their work I created [another pull-requests](https://github.com/pcloudcom/console-client/pull/133) which doesn't break the client for US clients but uses an environment variable to switch between both regions.
> 
> The best solution would of cause having a cli parameter, but to add this I needed more time to get into the code.

Seems like a suitable solution, as long as it doesn't break the US experience.

**Comment by ChuckNorrison** [comment]
same here

**Comment by banzoola** [comment]
looks like pclsync_lib.cpp has been updated with "bineapi.pcloud.com" already however this is still an issue for me.
